tests/ifsbench/ifsbench_run.py

```diff
@@ -0,0 +1,366 @@
+
+    tech_setup = TechSetup(
+        data_handlers_runtime = data_handlers_runtime,
+        env_handlers = env_handlers
+    )
+
+    return Benchmark(science = science_setup, tech=tech_setup)
+
+
+@cli.command('from_yaml', context_settings={"auto_envvar_prefix": "IFSBENCH"})
+@click .argument('yaml-path', type=click.Path(exists=True))
+@click.argument('science', type=str)
+@click.option('--binary-path', type=click.Path(exists=True),
+              help='Path to the ecLand binary')
+@click.option('--tech', type=str, default='default',
+              help='Specify the used technical setup')
+@click.option('--run-dir', type=click.Path(), default=None,
+              help='Run directory for the tests (temporary directory by default)')
+@click.option('--tasks', type=int, default=None,
+              help='Number of tasks to run')
+@click.option('--threads', type=int, default=None,
+              help='Number of threads to use')
+@click.option('--arch', default=None, type=str,
+              help='The architecture to use.')
+@click.option('--launcher-flags', default=[], multiple=True, type=str,
+              help='Additional flags that are passed to the launcher')
+@click.option('--validate', 'validate_path', type=click.Path(exists=True),
+              help='Validate results against given result file.')
+def from_yaml(yaml_path, science, tech, binary_path, run_dir, tasks, threads,
+    arch, launcher_flags, validate_path):
+    """
+    Run ecland benchmark using a YAML configuration file.
+
+    Params
+    ------
+    yaml-path:
+        The path to the configuration YAML file.
+    science:
+        Which science (=experiment) to run. Must exist in the `science` entry
+        of the YAML file.
+    """
+    yaml_path = Path(yaml_path).resolve()
+
+    with yaml_path.open('r', encoding='utf-8') as f:
+        yaml_data = yaml.safe_load(f)
+
+    # Load the full configuration from the YAML file.
+    ecland_config = EclandConfig.from_config(yaml_data)
+
+    # Select the scientific and technical setup.
+    science_input = ecland_config.science[science]
+    tech_input = ecland_config.tech[tech]
+
+    if binary_path:
+        science_input.binary_path = Path(binary_path).resolve()
+
+    benchmark = build_ecland_benchmark(science=science_input, tech=tech_input)
+
+    job = science_input.job
+    if tasks:
+        job.tasks = tasks
+    if threads:
+        job.cpus_per_task = threads
+
+    arch = ecland_config.arch.get(arch, ecland_config.arch['default'])
+
+    if run_dir:
+        run_dir = Path(run_dir).resolve()
+        context = nullcontext(run_dir)
+    else:
+        context = TemporaryDirectory(dir=Path.cwd())
+
+    with context as run_dir:
+        run_dir = Path(run_dir)
+        benchmark.setup_rundir(run_dir)
+
+        bench_result = benchmark.run(
+            run_dir=run_dir,
+            job=job,
+            arch=arch,
+            launcher_flags=launcher_flags
+        )
+
+        result = EclandResult.from_rundir(
```

Comment:
Is the following the same logic as the validate function below? Could this be simply called from here? Just a thought, not necessarily required.